lib/Types.cpp

```diff
@@ -214,42 +217,70 @@ Type *ExtractSmallerStructField(const DataLayout &DL, Type *Ty) {
   return Smaller;
 }
 
+// Returns a preferred type chosen from two types.
+// In particular:
+// - If either is nullptr, return the other.
+// - When one is an aggregate or vector whose first nested member at some
+//   level of nesting "matches" the other, then return the aggregate or vector.
+//   They both have the same machine byte address, but we prefer the larger
+//   type.  Here two types "match" if they're the same, or if they are
+//   scalar numeric types (integer or float0 with the same bit size.
+// - If one type is a struct and is larger than the other type, but its
+//   smallest member is smaller than the other type, then return
+//   that member.
+// - Otherwise, if one type is smaller than the other, return the smaller one.
+// - When the sizes are the same, prefer the packed structure, if it exists.
 Type *SmallerTypeNotAliasing(const DataLayout &DL, Type *TyA, Type *TyB) {
```

Comment:
This method is possibly  misnamed.  Perhaps it should be PreferredTypeNotAliasing or something like that.

It's not always the smaller type.